### PR TITLE
feat: add ability to define flavour for tag

### PR DESCRIPTION
**Pull request by A1994SC**
## Problem Statement

At the moment which flavour of image tag used can only be set by pinning the image tag to a certain version:

```yaml
image:
  tag: "v0.9.9-ubi"
```

If I wanted to a system like fluxcd to automatically update the chart for me, I would need to have the [image automation system ](https://fluxcd.io/flux/components/image/) installed or to manually update the helm chart crd to keep the oci image in line with the chart.

## Related Issue

N/A

## Proposed Changes

Add an option field under `image` / `webhook.image` / `certController.image` called `flavour` to allow users to use the `ubi` or `ubi-boringssl` versions.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
hi @A1994SC! What happens if we introduce a breaking change to the helm chart? 🤔 IIRC, we always pin the image version anyways to prevent that, so I'm failing to see how that would really work.

**Comment by A1994SC** [comment]
Well this will still default to using the chart app version regardless of you chose to use a `flavour` or not. And you can still use the `image.tag` as how you chose your flavour. This just allows you to decouple the flavour from image/chart version.

**Comment by gusfcarvalho** [comment]
Can you add some unit tests to it? :) I’d like to see at least one regression test (no flavor) and one test setting up `flavour: ubi`

**Comment by A1994SC** [comment]
> Can you add some unit tests to it? :) I’d like to see at least one regression test (no flavor) and one test setting up `flavour: ubi`

Sure, I will create some this evening. I am unsure where the unit tests are, do you mind pointing me in the right direction? Thank you.

**Comment by A1994SC** [comment]
@gusfcarvalho I added unit tests

File: deploy/charts/external-secrets/README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ The command removes all the Kubernetes components associated with the chart and
 | certController.image.pullPolicy | string | `"IfNotPresent"` |  |
 | certController.image.repository | string | `"ghcr.io/external-secrets/external-secrets"` |  |
 | certController.image.tag | string | `""` |  |
+| certController.image.falvour | string | `""` | The falvour of the image tag to use. The default is distroless. |
```
Comment (gusfcarvalho) [inline deploy/charts/external-secrets/README.md:50]:
```suggestion
| certController.image.flavour| string | `""` | The flavour of the image tag to use. The default is distroless. |
```

Comment (gusfcarvalho) [reply deploy/charts/external-secrets/README.md:50]: I think you can use `make helm.docs` to generate this update to the chart README- this way this is done programatically and we avoid typos.

File: deploy/charts/external-secrets/values.yaml [hunk 2]
```diff
@@ -7,10 +7,12 @@ image:
   repository: ghcr.io/external-secrets/external-secrets
   pullPolicy: IfNotPresent
   # -- The image tag to use. The default is the chart appVersion.
+  tag: ""
+  # -- The flavour of tag you want to use
   # There are different image flavours available, like distroless and ubi.
   # Please see GitHub release notes for image tags for these flavors.
   # By default the distroless image is used.
-  tag: ""
+  flavour: ""
```
Comment (gusfcarvalho) [inline deploy/charts/external-secrets/values.yaml:15]: In here you can add comments to the top, this way `make helm.docs` will generate the appropriate text to `charts/README.md`

Comment (A1994SC) [reply deploy/charts/external-secrets/values.yaml:15]: Updated docs, I was using podman and it caused some permission issues, but it is all good now

Comment (A1994SC) [reply deploy/charts/external-secrets/values.yaml:15]: @gusfcarvalho sorry to at you but did you have any other comments?

**Review by moolen — APPROVED** [review]
Hey @A1994SC, thank you for your contribution, this is a really great UX improvement :bow: :tada:

**Comment by A1994SC** [comment]
@moolen thanks! Glad to help how I can!

**Pull request merged**